experimental/LinearQuotients/src/misc.jl

```diff
@@ -1,29 +1,40 @@
-# Try to 'update' the base_ring of G
-function map_entries(K::Field, G::MatrixGroup)
-  g = dense_matrix_type(K)[]
-  for h in gens(G)
-    push!(g, map_entries(K, h.elm))
-  end
-  return matrix_group(g)
-end
+@doc raw"""
+    is_pseudo_reflection(g::MatrixGroupElem)
 
-function is_reflection(g::MatrixGroupElem)
+Return `true` if `g` is a pseudo-reflection, `false` otherwise. By a
+pseudo-reflection, we mean a matrix with fixed space of codimension 1.
+"""
+function is_pseudo_reflection(g::MatrixGroupElem)
   return rank(g.elm - one(parent(g)).elm) == 1
 end
 
-function subgroup_of_reflections(G::MatrixGroup)
+@doc raw"""
+    subgroup_of_pseudo_reflections(G::MatrixGroup)
+
+Return the subgroup `H` of `G` generated by the pseudo-reflections in `G` and an
+embedding `H \to G`.
+
+See also [`is_pseudo_reflection`](@ref).
+"""
+function subgroup_of_pseudo_reflections(G::MatrixGroup)
   g = elem_type(G)[]
   for c in conjugacy_classes(G)
-    if is_reflection(representative(c))
+    if is_pseudo_reflection(representative(c))
       append!(g, collect(c))
     end
   end
-  return matrix_group(base_ring(G), degree(G), g)
+  return sub(G, g, check = false)
 end
 
-# Check if G contains reflections
+@doc raw"""
+    is_small_group(G::MatrixGroup)
```

Comment:
This name is also a bit problematic, seeing as we have the "small groups" library, but it can be changed at another point.